src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -367,475 +326,520 @@ private void AnalyzeNodeForDtdProcessingOverloads(SyntaxNodeAnalysisContext cont
                 }
             }
 
-            private void AnalyzeNodeForDtdProcessingProperties(SyntaxNodeAnalysisContext context)
+            private void AnalyzeFieldDeclaration(OperationAnalysisContext context)
             {
-                SyntaxNode node = context.Node;
-                SemanticModel semanticModel = context.SemanticModel;
+                IFieldInitializer fieldInit = context.Operation as IFieldInitializer;
 
-                SyntaxNode lhs = _syntaxNodeHelper.GetAssignmentLeftNode(node);
-                if (lhs == null)
+                if(fieldInit == null)
                 {
                     return;
                 }
 
-                IPropertySymbol property = SyntaxNodeHelper.GetCalleePropertySymbol(lhs, semanticModel);
-                if (property == null)
+                foreach (IFieldSymbol field in fieldInit.InitializedFields)
+                {
+                    IOperation valueOperation = fieldInit.Value;
+                    AnalyzeObjectCreationInternal(context, field, valueOperation);
+                }
+            }
+
+            private void AnalyzeObjectCreationInternal(OperationAnalysisContext context, ISymbol variable, IOperation value)
+            {
+                IObjectCreationExpression objCreation = value as IObjectCreationExpression;
+
+                if (objCreation == null)
                 {
                     return;
                 }
 
-                if (property.MatchPropertyDerivedByName(_xmlTypes.XmlDocument, SecurityMemberNames.InnerXml))                                       //FxCop CA3058
+                if (_objectCreationOperationsAnalyzed.Contains(objCreation))
```

Comment:
ObjectCreation can be in a field declaration, can be in a variable declaration, or in a temporary object. If this is a variable declaration, this function will end up get invoked by both VariableDeclarationExpression and ObjectCreationExpression operation even though 1 is child of the other. this function this is just making sure we haven't seen this expression from somewhere else.